libs/core/src/lib/overflow-layout/overflow-layout.service.ts

```diff
@@ -149,16 +150,14 @@ export class OverflowLayoutService implements OnDestroy {
             item.index = index;
             item.first = index === 0;
             item.last = index === allItems.length - 1;
+            visibleContainerItems[index].containerRef.hidden = false;
         });
 
         this._detectChanges$.next();
 
         allItems = this.config.direction === 'right' ? allItems : allItems.reverse();
-        const visibleContainerItems =
-            this.config.direction === 'right'
-                ? this.config.visibleItems.toArray()
-                : this.config.visibleItems.toArray().reverse();
-        visibleContainerItems.forEach((i) => (i.containerRef.hidden = false));
+        visibleContainerItems =
+            this.config.direction === 'right' ? visibleContainerItems : visibleContainerItems.reverse();
```

Comment:
Never noticed this earlier, but would not it be more correct to use terms "start" and "end" instead of "left" and "right"? We support RTL